pkg/commands/deploy_actions.go

```diff
@@ -0,0 +1,1012 @@
+package commands
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"math/big"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/Layr-Labs/crypto-libs/pkg/bn254"
+	"github.com/Layr-Labs/crypto-libs/pkg/keystore"
+	"github.com/Layr-Labs/devkit-cli/config/configs"
+	"github.com/Layr-Labs/devkit-cli/config/contexts"
+	"github.com/Layr-Labs/devkit-cli/pkg/common"
+	"github.com/Layr-Labs/devkit-cli/pkg/common/iface"
+	allocationmanager "github.com/Layr-Labs/eigenlayer-contracts/pkg/bindings/AllocationManager"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/urfave/cli/v2"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+)
+
+type DeployContractTransport struct {
+	Name    string
+	Address string
+	ABI     string
+}
+
+type DeployContractJson struct {
+	Name    string      `json:"name"`
+	Address string      `json:"address"`
+	ABI     interface{} `json:"abi"`
+}
+
+func StartDeployL1Action(cCtx *cli.Context) error {
+	// Get logger
+	logger := common.LoggerFromContext(cCtx.Context)
+	caser := cases.Title(language.English)
+
+	// Extract vars
+	contextName := cCtx.String("context")
+	useZeus := cCtx.Bool("use-zeus")
+
+	// Migrate config
+	configsMigratedCount, err := configs.MigrateConfig(logger)
+	if err != nil {
+		logger.Error("config migration failed: %w", err)
+	}
+	if configsMigratedCount > 0 {
+		logger.Info("configs migrated: %d", configsMigratedCount)
+	}
+
+	// Migrate contexts
+	contextsMigratedCount, err := contexts.MigrateContexts(logger)
+	if err != nil {
+		logger.Error("context migrations failed: %w", err)
+	}
+	if contextsMigratedCount > 0 {
+		logger.Info("contexts migrated: %d", contextsMigratedCount)
+	}
+
+	// Load config for selected context
+	config, err := common.LoadConfigWithContextConfig(contextName)
+	if err != nil {
+		return err
+	}
+	if contextName == "" {
+		contextName = config.Config.Project.Context
+	}
+
+	// Check for context
+	yamlPath, rootNode, contextNode, err := common.LoadContext(contextName)
+	if err != nil {
+		return fmt.Errorf("context loading failed: %w", err)
+	}
+
+	// Fetch EigenLayer addresses using Zeus if requested
+	if useZeus {
+		err = common.UpdateContextWithZeusAddresses(cCtx.Context, logger, contextNode, contextName)
+		if err != nil {
+			logger.Warn("Failed to fetch addresses from Zeus: %v", err)
+			logger.Info("Continuing with addresses from config...")
+		} else {
+			logger.Info("Successfully updated context with addresses from Zeus")
+
+			// Write yaml back to project directory
+			if err := common.WriteYAML(yamlPath, rootNode); err != nil {
+				return fmt.Errorf("failed to save updated context: %v", err)
+			}
+		}
+	}
+
+	// Get fork_urls for the provided network (default to sepolia testnet)
+	l1ForkUrl, err := common.GetForkUrlDefault(contextName, config, common.L1)
+	if err != nil {
+		return fmt.Errorf("L1 fork URL error %w", err)
+	}
```

Comment:
why is this blocking? it shouldn't be required here, right?

Reply:
We use the `fork_url` as the `rpc_url` here, I'll tidy things up so that we're accepting either, but all of our documentation is based around `fork_url`s at the minute (again, I'll get this tidied up)